src/ngraph/distributed.hpp

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#ifdef IN_NGRAPH_LIBRARY
+#error("distributed.hpp is for external use only")
+#endif
+
+#include <mpi.h>
```

Comment:
do we need distributed flag here during compile?

Reply:
Not really. This file is used externally.

Reply:
Sorry I don't understand. Could you please elaborate how it's used, e.g. for MXNet?

Reply:
The bridge will `#include` this header file, not nGraph. So, it depends on what distributed flag the bridge uses. If MXNet uses `MXNET_NGRAPH_DISTRIBUTED_ENABLE` and TF uses `NGRAPH_TF_DISTRIBUTED`, then having `#ifdef NGRAPH_DISTRIBUTED #include <mpi.h> #endif` in this file will not work, since the bridge doesn't know about `NGRAPH_DISTRIBUTED`. Every bridge has its own style for defining these flags, so it's hard to unify.